core/src/main/java/org/elasticsearch/cluster/metadata/IndexMetaData.java

```diff
@@ -383,6 +383,14 @@ public MappingMetaData mapping(String mappingType) {
         return mappings.get(mappingType);
     }
 
+    public static final Setting<String> INDEX_MERGE_SOURCE_UUID = Setting.simpleString("index.merge.source.uuid");
```

Comment:
don't we need an IndexScope here?

Nit: `index.merge...` sounds to me like has something to do with the merge policy, but this is totally different. I'll try to come up with a suggestion and report back.